### ISSUE TITLE
Incorrect warning: "truthy value should be one of [false, true]"

### ISSUE THREAD

**Issue by uvsmtid** [adrienverge/yamllint#744]
# Problem

If you run `yamllint` against this file:

```yaml
---

on:
    whatever: "whatever"
```

```sh
yamllint test.yaml
```

It generates this warning:

```
test.yaml
  3:1       warning  truthy value should be one of [false, true]  (truthy)
```

Apparently, it complains about `on:` dict key, not dict value.

There should not be any warning for dict keys.

For example, [`on:` is a top level keys for GitHub Workflow configs](https://github.com/argrelay/argrelay/blob/32b1f2d69bf8ad159329dca003895ea57c4489e7/.github/workflows/argrelay.bootstrap.yaml#L3).

# Versions

```
yamllint==1.37.0
```

**Comment by Timmmm** [comment]
Maybe the fact that so many people reported this is a sign that something should be changed in `yamllint`. I can think of two easy solutions:

1. Disable `truthy: check-keys` by default. This is a linter so it doesn't matter too much if it misses a rare mistake.
2. Add a link or some text in the error message for truthy checks when they fail for keys that tells you what to do.

I.e. instead of

```
 3:1       error    truthy value should be one of [false, true]  (truthy)
```

Have

```
 3:1       error    truthy key should be one of [false, true]; to disable this check for keys see https://..../  (truthy)
```
